sql/core/src/test/scala/org/apache/spark/sql/execution/ui/SQLListenerSuite.scala

```diff
@@ -343,6 +343,8 @@ class SQLListenerMemoryLeakSuite extends SparkFunSuite {
       .set("spark.sql.ui.retainedExecutions", "50") // Set it to 50 to run this test quickly
     val sc = new SparkContext(conf)
     try {
+      // Clear the sql listener created by a previous test suite.
+      SQLContext.clearSqlListener()
```

Comment:
Many unit tests use `sqlContext.listener`. Can you please suggest how to update the unit tests if we changed to use an `AtomicBoolean`?